spec/factories/topics.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+# == Schema Information
+#
+# Table name: topics
+#
+#  id         :bigint           not null, primary key
+#  name       :string           not null
+#  created_at :datetime         not null
+#  updated_at :datetime         not null
+#
+# Indexes
+#
+#  index_topics_on_name  (name) UNIQUE
+#
+FactoryBot.define do
+  factory :topic do
```

Comment:
According to the FactoryBot best practices:

> It is recommended that you have one factory for each class that provides the simplest set of attributes necessary to create an instance of that class

https://github.com/thoughtbot/factory_bot/blob/master/GETTING_STARTED.md#best-practices

I'm wondering if the image is actualy part of "the simplest set of attributes".

Maybe we can do something like this:
```ruby
FactoryBot.define do
  factory :topic do
    name { Faker::Lorem.word }

    trait :with_image do
      transient do
        sequence(:tmp_img) { |n| Tempfile.new(["tmp_img_#{n}", 'jpg']).path }
      end

      image { Rack::Test::UploadedFile.new(tmp_img, 'image/png') }
    end
  end
end
```

So then
```ruby
let(:topic) { create(:topic) } # It only has a name

let(:topic) { create(:topic, :with_image) } # It has name and image
```

Reply:
Good one, I will update this factory! Thanks! @JulianPasquale